### PR TITLE
KYLIN-4843 Support INTERSECT_COUNT/INTERSECT_VALUE function for Kylin 4

### PR DESCRIPTION
**Pull request by zzcclp**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [x] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zzcclp** [comment]
Run test cases local successfully.

**Comment by hit-lacus** [comment]
For expected result, could you please use human-readable format like csv but not parquet?

**Comment by zzcclp** [comment]
> For expected result, could you please use human-readable format like csv but not parquet?

parquet results is only for 'intersect_value' function test cases, because the returned results of 'intersect_value' function is array type and csv can not support array type.

**Comment by hit-lacus** [comment]
After that, I find that checkstyle plugin failed, please try check again .

image

**Comment by zzcclp** [comment]
> > For expected result, could you please use human-readable format like csv but not parquet?
> 
> parquet results is only for 'intersect_value' function test cases, because the returned results of 'intersect_value' function is array type and csv can not support array type.

changes the returned datatype of 'intersect_value' function to StringType, so  use csv files as expected results format.

File: kylin-spark-project/kylin-spark-query/src/main/scala/org/apache/spark/sql/SparkOperation.scala [hunk 1]
```diff
@@ -21,7 +21,6 @@ import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.plans.logical.GroupingSets
 import org.apache.spark.sql.types.StructType
-import org.apache.spark.sql.{Column, DataFrame, Row, SparderContext}
```
Comment (hit-lacus) [inline kylin-spark-project/kylin-spark-query/src/main/scala/org/apache/spark/sql/SparkOperation.scala:24]: Why remoe this import?

Comment (zzcclp) [reply kylin-spark-project/kylin-spark-query/src/main/scala/org/apache/spark/sql/SparkOperation.scala:24]: now move 'SparkOperation' to package 'org.apache.spark.sql', so don't need to import the classes which are in the same package.

**Review by hit-lacus — APPROVED** [review]
Approved

**Pull request merged**